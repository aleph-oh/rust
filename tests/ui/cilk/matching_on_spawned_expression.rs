#![feature(cilk)]
// Tests that matching on a spawned expression gives an error.

fn main() {
    let x: Option<usize> = cilk_spawn { None };
    match x {
//~^ ERROR used binding `x` is possibly-uninitialized [E0381]
        Some(x) => {}
        None => {}
    }
}
