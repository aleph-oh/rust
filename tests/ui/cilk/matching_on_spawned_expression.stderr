error[E0381]: used binding `x` is possibly-uninitialized
  --> $DIR/matching_on_spawned_expression.rs:6:11
   |
LL |     let x: Option<usize> = cilk_spawn { None };
   |         -                             -------- binding initialized here in some conditions
   |         |
   |         binding declared here but left uninitialized
LL |     match x {
   |           ^ `x` used here but it is possibly-uninitialized

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0381`.
